datafusion/core/src/execution/context/mod.rs

```diff
@@ -790,9 +790,26 @@ impl SessionContext {
         }
 
         if exist {
-            match cmd.if_not_exists {
-                true => return self.return_empty_dataframe(),
-                false => {
+            match (cmd.if_not_exists, cmd.or_replace) {
```

Comment:
I think it makes sense to make this behavior consistent with `create_memory_table`